### ISSUE TITLE
Upgrade NGINX to version 1.20.1 or higher - another follow-up

### ISSUE THREAD

**Issue by Lisenish** [aws/elastic-beanstalk-roadmap#221]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave "+1" or "me too" comments. They generate extra noise for issue followers and do not help prioritize the request.

**Tell us about your request**
Upgrade Nginx

**Is this request specific to an Elastic Beanstalk [platform](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platforms-supported.html)?**
not platform specific. affects all beanstalks

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Our pentests are reporting an vulnerability in the current nginx version.

**Are you currently working around this issue?**
There is a workaround provided by @goodthang
https://github.com/aws/elastic-beanstalk-roadmap/issues/194#issuecomment-1016692947

**Additional context**
It's already 3rd issue about this problem, the other one was closed without an actual fix, I don't know why. 
I've created another issue because it seems that comments in the closed issues are ignored.

https://github.com/aws/elastic-beanstalk-roadmap/issues/194

**Comment by 1tonyca** [comment]
@Lisenish, thanks for the follow-up.

If your question is whether [CVE-2021-23017](https://www.nginx.com/blog/updating-nginx-dns-resolver-vulnerability-cve-2021-23017) has already been addressed in the version of nginx on your beanstalk instance, and the answer is yes if you are using any platform versions since July, 2021. 

You can confirm that by running the following cmd on your beanstalk instance, and look for Release that is 2.amzn2.0.3 or newer, per [Amazon Linux team](https://forums.aws.amazon.com/thread.jspa?messageID=990394)

$ yum info nginx
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
Installed Packages
Name        : nginx
Arch        : x86_64
Epoch       : 1
Version     : 1.20.0
Release     : 2.amzn2.0.4
Size        : 1.7 M
Repo        : installed
From repo   : amzn2extra-nginx1

Beanstalk platform consumes nginx directly from amazon linux extra repo, as indicated from the yum info above. The package from Amazon Linux extra repo does not version the same as the package generated by nginx.org. 

Please do use yum info on Beanstalk instance to confirm if your nginx version is current.

**Comment by goodthang** [comment]
@Lisenish - Yes, we use the work-around solution as I outlined in the [comment](https://github.com/aws/elastic-beanstalk-roadmap/issues/194#issuecomment-1016692947).  That's the only work-around I found to be suitable for us.
@1tonyca - The Beanstalk platform uses Linux extra repo, and that repo (last time I checked, Oct 2021) did not have nginx 1.20.1.  At the time, none of the repos had nginx 1.20.1.

I think the appropriate solution is to have nginx 1.20.1 in the Amazon Linux extra repo that Beanstalk can pull from.

**Comment by Lisenish** [comment]
@1tonyca This [documentation page](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platforms-supported.html#platforms-supported.nodejs) also says that it's **1.20.0** still or this table also refers to some internal version which doesn't correspond to the usual ngnix versioning? How I can map oficial nginx versions to the version from the extra linux repo?

The thing is that server header is also returning nginx 1.20 version and it triggers pentest checkers, you can see that I'm not the only one who have an issue with it.

**Comment by 1tonyca** [comment]
Beanstalk platform used the nginx that was built and published by amazon linux, and amazon linux did not follow the same versioning strategy as nginx.org. 

In terms of vulnerabilities introduced by CVE-2021-23017, there were no issues with the version that was built by Amazon Linux, as they [clarified last year](https://forums.aws.amazon.com/thread.jspa?messageID=990394).

If there is a need to pass any checkers that only validate version number, please use the .ebextension workaround for now.

**Comment by Lisenish** [comment]
@1tonyca Yeah I got it that ELB uses its own versions I just wanted to point out that it's confusing in the documentation now and probably it's worth clarifying it somewhere and providing some kind of mapping between official nginx.org versions and the ones ELB is using.

But yeah the main question is solved, I'm closing it thank you for the clarification.

**Comment by bsplosion** [comment]
The forums thread link in the comment from February is dead now (apparently it's archived but not available after migration to re:Post).  Is there any information about mapping AML2 nginx versions to the published nginx.org versions?  It seems nonsensical that they'd be so incredibly close in version numbers yet not match.

Also, is there any documentation published anywhere at all that Amazon is doing their own version of nginx?  I can't seem to find any info about this anywhere and it came as a surprise reading through this issue.  Are there other compromises or caveats with this special AML2 nginx version that should be disclosed?

**Comment by VaguelyOnline** [comment]
After hours of searching for an answer of how to upgrade the Nginx of my elastic beanstalk image for my PHP Laravel application - I have found this thread. We too ran a pen test and had the version of nginx flag this as a potential issue. It also flagged a number of other issues concerning OpenSSH, including the following CVEs. It seems that most would be fixed with an update to the OpenSSH package being used.

But if the Nginx package used doesn't track the official release version, perhaps these issues have already been addressed? After all, the openssh package version indicates it's also a special build for AWS. How can anyone tell if a given CVE has been addressed in Elastic Beanstalk image / Amazon Linux?

$ yum info openssh
Name        : openssh
Arch        : aarch64
Version     : 7.4p1
Release     : 22.amzn2.0.1

CVE's flagged relating to Nginx and OpenSSH. Note - this is using the latest (as per date of writing) PHP elastic beanstalk images.

nginx 0.6.18 - 1.20.0 1-byte Memory Overwrite Vulnerability CVE-2021-23017
OpenSSH <= 8.6 Command Injection Vulnerability  CVE-2021-23017
Diffie-Hellman Ephemeral Key Exchange DoS Vulnerability (SSH, D(HE)ater)    CVE-2002-20001
nginx <= 1.21.1 Information Disclosure Vulnerability    CVE-2013-0337
OpenSSH 6.2 <= 8.7 Privilege Escalation Vulnerability   CVE-2021-41617
OpenBSD OpenSSH <= 7.9 Multiple Vulnerabilities CVE-2018-20685, CVE-2019-6109, CVE-2019-6110, CVE-2019-6111
OpenBSD OpenSSH Information Disclosure Vulnerability (CVE-2020-14145)   CVE-2020-14145
SSL/TLS: BREACH attack against HTTP compression CVE-2013-3587
OpenSSH 'auth2-gss.c' User Enumeration Vulnerability - Linux    CVE-2018-15919
OpenSSH 'sftp-server' Security Bypass Vulnerability (Linux) CVE-2017-15906
OpenSSH < 7.8 User Enumeration Vulnerability - Linux    CVE-2018-15473
OpenSSH Information Disclosure Vulnerability (CVE-2016-20012)   CVE-2016-20012
